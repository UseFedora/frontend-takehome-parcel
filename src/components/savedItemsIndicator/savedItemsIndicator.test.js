import React from 'react';
import { shallow } from 'enzyme';
import SavedItemsIndicator from './';

describe('SavedItemsIndicator', () => {
  it('should render correctly', () => {
    const component = shallow(<SavedItemsIndicator />);
    expect(component).toMatchSnapshot();
  });

  it('should display the proper grammatical label for 1 item', () => {
    const component = shallow(
      <SavedItemsIndicator
        numberOf={1}
      />
    );
    expect(component.text()).toMatch(/1 gem$/);
  });

  it('should display the proper grammatical label for 0 items', () => {
    const component = shallow(
      <SavedItemsIndicator
        numberOf={0}
      />
    );
    expect(component.text()).toMatch(/0 gems$/);
  });

  it('should display the proper grammatical label for more than 1 items', () => {
    const component = shallow(
      <SavedItemsIndicator
        numberOf={14}
      />
    );
    expect(component.text()).toMatch(/14 gems$/);
  });
});
