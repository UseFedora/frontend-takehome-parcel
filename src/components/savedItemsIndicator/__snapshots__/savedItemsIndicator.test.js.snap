// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SavedItemsIndicator should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SavedItemsIndicator
    numberOf={0}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <p>
        <FaRegGem />
        <span>
          0 gems
        </span>
      </p>,
      "className": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <FaRegGem />,
          <span>
            0 gems
          </span>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "0 gems",
            "className": undefined,
          },
          "ref": null,
          "rendered": "0 gems",
          "type": "span",
        },
      ],
      "type": "p",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <p>
          <FaRegGem />
          <span>
            0 gems
          </span>
        </p>,
        "className": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <FaRegGem />,
            <span>
              0 gems
            </span>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "className": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "0 gems",
              "className": undefined,
            },
            "ref": null,
            "rendered": "0 gems",
            "type": "span",
          },
        ],
        "type": "p",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
